### PR TITLE
feat: enable v1 root drf browsable and fix staticfiles

### PR DESCRIPTION
**Pull request by Alex-Izquierdo**
Enable the v1 root for DRF browsable and adds the missing collecstatic step. 
It requires updates in the installer/operator/ui_image to run the colllecstatic command and serve those static files. 
Jira: https://issues.redhat.com/browse/AAP-25508
Screenshot from 2024-06-18 18-42-08

### CODE REVIEW

**Review by rooftopcellist — DISMISSED** [review]

**Review by mkanoor — DISMISSED** [review]

**Comment by Alex-Izquierdo** [comment]
Don't merge this for now, please. I want to sync the required changes in other places like the installer to avoid broken builds.

**Review by mkanoor — APPROVED** [review]

File: tests/integration/api/test_v1_root.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                "/event-streams/",
+                "/users/me/awx-tokens/",
+                "/credential-types/",
+                "/eda-credentials/",
+                "/decision-environments/",
+                "/organizations/",
+                "/teams/",
+            ],
+            True,
+            id="with_shared_resource",
```
Comment (jshimkus-rh) [inline tests/integration/api/test_v1_root.py:48]: Is there a purpose to `id`?  There's not a third entry in the parameter list.

**Review by hsong-rh — APPROVED** [review]

File: tests/integration/api/test_v1_root.py (hunk 1 above)
Comment (Alex-Izquierdo) [reply tests/integration/api/test_v1_root.py:48]: @jshimkus-rh 
`id` is a parameter for pytest. It's the name of the test case. 
```
tests/integration/api/test_v1_root.py::test_v1_root[with_shared_resource] PASSED [ 45%]
tests/integration/api/test_v1_root.py::test_v1_root[no_shared_resource] PASSED [ 45%]
```

Comment (jshimkus-rh) [reply tests/integration/api/test_v1_root.py:48]: Learn something new everyday.  I guess I never looked that close at the output from stuff I parameterized.

**Review by jshimkus-rh — APPROVED** [review]

**Review by Dostonbek1 — CHANGES_REQUESTED** [review]
Looking at the JIRA ticket expected behavior, this PR doesn't seem to consider the following case:
```
/api/eda should provide version options for the API
```

**Comment by Alex-Izquierdo** [comment]
> Looking at the JIRA ticket expected behavior, this PR doesn't seem to consider the following case:
> 
> ```
> /api/eda should provide version options for the API
> ```

It is out of the scope of this PR. As you can see in the description of the jira, only includes the v1/

**Review by Dostonbek1 — APPROVED** [review]

**Comment by Alex-Izquierdo** [comment]
> Another minor comment, it seems a little redundant to have the hostname before every endpoint listed in v1 root view, e.g. `http://localhost:8000/api/eda/users/me/`. Since it is obvious what the hostname is, maybe we can drop it; it would make the endpoints more readable too. Thoughts?

It seems to me a valid point. I think it is due to the way I generate automatically all the links. I was trying to merge this PR in sync with other required changes to avoid breaking build. New changes in the PR would complicate this. Do you agree to merge as it is and later address the host in a later PR?

**Pull request merged**